### PR TITLE
Models & migrations homework

### PR DESCRIPTION
**Pull request by motivatedHamster**

### CODE REVIEW

File: app/models/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Classroom < ActiveRecord::Base
+  validates :name, presence: true, uniqueness: true, length: { maximum: 255, too_long: "%{count} characters maximum" }
+  has_many :drills, dependent :destroy
+  has_many :drill_types, dependent :destroy
```
Comment (mel-quark) [inline app/models/classroom.rb:6]: i think you might mean 
`has_many :drill_types, through: drills`
http://guides.rubyonrails.org/association_basics.html#the-has-many-through-association

Comment (mel-quark) [inline app/models/classroom.rb:6]: though, actually, i don't think you need this connection...

File: app/models/drill_type.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class DrillType < ActiveRecord::Base
+  validates :name, presence: true, uniqueness: true, length: { maximum: 255, too_long: "%{count} characters maximum" }
+  validates :required_frequency, presence: true, inclusion: { in: %w(weekly monthly yearly quarterly twice_yearly), message: "%{value} must be one of: weekly, monthly, yearly, quarterly, twice_yearly" }
+  has_many :drills
+  belongs_to :classroom
```
Comment (mel-quark) [inline app/models/drill_type.rb:5]: not sure i understand why a drill type has a classroom...?

File: db/migrate/20150504224656_create_drill_types.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreateDrillTypes < ActiveRecord::Migration
+  def change
+    create_table :drill_types do |t|
+      t.string :name, limit: 255, null: false
+      t.string :required_frequency, limit: 255, null: false
+      t.references :classroom, index: true
```
Comment (mel-quark) [inline db/migrate/20150504224656_create_drill_types.rb:6]: you won't need this line

File: db/migrate/20150504224656_create_drill_types.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+      t.timestamps null: false
+    end
+    add_foreign_key :drill_types, :classrooms
```
Comment (mel-quark) [inline db/migrate/20150504224656_create_drill_types.rb:10]: cut this one, too

**Comment by mel-quark** [comment]
very nicely done =) just a few comments above.

File: app/models/classroom.rb (hunk 1 above)
Comment (motivatedHamster) [inline app/models/classroom.rb:6]: I wanted some way to represent the functionality where the frequency of a drill type is specific to a classroom.  One of the classes only has to perform fire drills once per quarter while the rest of the classes must have these drills monthly.

File: app/models/drill_type.rb (hunk 2 above)
Comment (motivatedHamster) [inline app/models/drill_type.rb:5]: this is the other side of the classroom drill_type relationship

**Comment by motivatedHamster** [comment]
Hi Mel, 

Looks like most or all of your comments are related to the drill_type classroom relationship.  I made this relationship to reflect the need for each classroom to have their own rules for drill frequency.  For example, one classroom does fire drills quarterly while other classes do fire drills monthly.  (It's based on what percentage of the time the class is occupied I think and could change.)

-j

Sent via the Ultra Pretentionator 4000™, an AT&T 4G LTE user tracking device.

<div>-------- Original message --------</div><div>From: Mel Quark notifications@github.com </div><div>Date:05/10/2015  11:24 PM  (GMT-08:00) </div><div>To: motivatedHamster/safety_drill_logger safety_drill_logger@noreply.github.com </div><div>Cc: motivatedHamster puzzle@gmail.com </div><div>Subject: Re: [safety_drill_logger] Models & migrations homework (#3) </div><div>
</div>very nicely done =) just a few comments above.

**Comment by mel-quark** [comment]
oh, i see. that's fine then. for UX, create a page with all the upcoming drills (include what classroom they're for, what kind of drill, etc). you'll need to seed data for this.

also, make it so you can filter on this page by teacher and drill type.  here are some gems that might be helpful: https://www.ruby-toolbox.com/categories/Table_Builders

**Pull request merged**